client/state/sites/guided-transfer/actions.js

```diff
@@ -90,13 +95,28 @@ export function saveHostDetails( siteId, data ) {
 			dispatch( receiveGuidedTransferStatus(
 				siteId, omit( response, '_headers' )
 			) );
+
+			dispatch( successNotice( translate( 'Thanks for confirming those details!' ) ) );
+
+			return response.host_details_entered === true;
 		};
 
-		const failure = error => dispatch( {
-			type: GUIDED_TRANSFER_HOST_DETAILS_SAVE_FAILURE,
-			siteId,
-			error,
-		} );
+		const failure = error => {
+			dispatch( {
+				type: GUIDED_TRANSFER_HOST_DETAILS_SAVE_FAILURE,
+				siteId,
+				error,
+			} );
+
+			if ( error.message ) {
+				dispatch( errorNotice( error.message ) );
+			} else {
+				dispatch( errorNotice( translate( `Whoops, there was an error saving your details. Please 
```

Comment:
Rather than dispatching the error notice here, I created a notices middleware for this purpose:

https://github.com/Automattic/wp-calypso/blob/master/client/state/notices/middleware.js

See #6645